vignettes/touchstone.Rmd

```diff
@@ -0,0 +1,121 @@
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+
+Start by initializing {touchstone} in your package repository with:
+```{r, eval = FALSE}
+touchstone::use_touchstone()
+```
+
+This will:
+
+- create a `touchstone` directory in the repository root with:
+
+  * `config.json` that defines how to run your benchmark. In particular, you can
+    define a `benchmarking_repo`, that is, a repository you need to run your bench mark 
+    (use `""` if you don't need an additional git repository checked out for your 
+    benchmark).
+    This repository will be cloned into `benchmarking_path`. For example to benchmark
+    {styler}, we format the {here} package with `style_pkg()` that is not part 
+    of the {styler} repository, but with the below config, will be located at
+    `touchstone/sources/here` during benchmarking. The code you want to 
+    benchmark comes from the benchmarked repository, which in our case is the root 
+    from where you call `use_touchstone()` and hence does not need to be defined 
+    explicitly.
+```json
+    { 
+      "benchmarking_repo": "lorenzwalthert/here", 
+      "benchmarking_ref": "ca9c8e69c727def88d8ba1c8b85b0e0bcea87b3f", 
+      "benchmarking_path": "touchstone/sources/here", 
+      "os": "ubuntu-18.04", 
+      "r": "4.0.0", 
+      "rspm": "https://packagemanager.rstudio.com/all/__linux__/bionic/291" 
+    }
+```
+  * `script.R`, the script that runs the benchmark.
+  * `header.R`, the script containing the default PR comment header, see `?touchstone::pr_comment`.
+  * `footer.R`, the script containing the default PR comment footer, see `?touchstone::pr_comment`.
+- add the `touchstone` directory to `.Rbuildignore`.
+- write the workflow files^[Note that these files must be committed to the default branch before
+{touchstone} continuous benchmarking will be triggered for new PRs.] you need to
+  invoke touchstone in new pull requests into `.github/workflows/`. 
+
+Now you just need to modify `script.R` to run the benchmarks you are interested in. 
+
+## script.R
+
+The file consists of three parts, two of which you will need to modify to fit your needs. Within the github action {touchstone} will always run the `script.R` from the `HEAD` branch, so you can modify or extend the benchmarks if needed to fit your current branch.
+
+### Setup
+
+We first install the different package versions in separate libraries with `refs_install()`, this is mandatory for any `script.R`. If you want to access some directory or file which only exists on one of the branches you can use `pin_assets(..., ref = "branch")` to make them usable across branches. You can of course also run arbitrary code to prepare for the benchmarks. 
+```{r}
+touchstone::refs_install() # installs branches to benchmark
+touchstone::pin_assets("data/all.Rdata", "inst/scripts") # pin files and directories
+
+load(path_pinned_asset("all.Rdata")) # perform other setup you need for the benchmarks
```

Comment:
I am not quite sure I understand, that's what it says?

We have separate dirs in the temp dir for head and base ref now that we allow pinning from both to avoid name collisions.

Reply:
> We have separate dirs in the temp dir for head and base ref now that we allow pinning from both to avoid name collisions.

Yes, but if I pin `data/all.Rdata`, it would be most natural to access it with `path_pinned_asset("data/all.Rdata")` instead of `path_pinned_asset("all.Rdata")`. Do you agree or not?